### PR TITLE
String operations

### PR DESCRIPTION
**Pull request by KTGypsy**
## What:

* String operations: cat, substr, in

## Why:

* To enable more features 

## Example:

* https://jsonlogic.com/operations.html#string-operations

### CODE REVIEW

File: src/commonTest/kotlin/operations/string/InTest.kt [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                    expression = mapOf("in" to listOf("t", listOf("team"))),
+                    data = emptyMap<String, Any>(),
+                    result = false
+                ),
+                TestInput(
+                    expression = mapOf("in" to listOf("1", 133)),
+                    data = emptyMap<String, Any>(),
+                    result = false
+                ),
+                TestInput(
```
Comment (PaulinaSadowskaAllegro) [inline src/commonTest/kotlin/operations/string/InTest.kt:45]: you can also add test for: 
```
{"in": [1, "122"]}
```
(returns true)

File: src/commonTest/kotlin/operations/string/InTest.kt [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                    expression = mapOf("in" to listOf(1, 133)),
+                    data = emptyMap<String, Any>(),
+                    result = false
+                ),
+                TestInput(
```
Comment (PaulinaSadowskaAllegro) [inline src/commonTest/kotlin/operations/string/InTest.kt:50]: this should return true:
```
{"in": [true, "true cat"]}
```

File: src/commonMain/kotlin/operations/string/Substr.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    private fun String.fromStartIndexToEnd(startIndex: Int) = if (startIndex > 0) {
+        substring(startIndex)
+    } else {
+        substring(length + startIndex)
+    }
+
+    private fun String.fromStartIndexToEndIndex(startIndex: Int, endIndex: Int) = when {
+        startIndex >= 0 && endIndex > 0 -> substring(startIndex, startIndex + endIndex)
```
Comment (PaulinaSadowskaAllegro) [inline src/commonMain/kotlin/operations/string/Substr.kt:36]: soo isn't `endIndex` in fact the length of substring we want to get, not endIndex?

**Review by PaulinaSadowskaAllegro — CHANGES_REQUESTED** [review]

File: src/commonMain/kotlin/operations/string/Substr.kt (hunk 3 above)
Comment (GypsyBox) [reply src/commonMain/kotlin/operations/string/Substr.kt:36]: ye, youre right. I renamed it in postprocess. It should be named more like charsCount

**Review by PaulinaSadowskaAllegro — DISMISSED** [review]

**Review by PaulinaSadowskaAllegro — APPROVED** [review]

**Review by mkrogulskiallegro2 — APPROVED** [review]

**Pull request merged**